lib/takeaway.rb

```diff
@@ -0,0 +1,37 @@
+require './lib/menu.rb'
+require './lib/sms_sender.rb'
+class Takeaway
+
+attr_reader :customer_order
+
+ def view_menu
+  Menu.new.menu
+ end
+
+ def place_order(order_hash, exact_total)
+  @customer_order = order_hash
+  order_checker
+  message = 'sorry your calculated sum does not match the order total'
+  raise(message) if order_sum != exact_total
+  sms_send
+ end
+
+private
+
+def order_checker
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for order_checker is too high. [16.4/15] (http://c2.com/cgi/wiki?AbcMetric)<br>Metrics/MethodLength: Method has too many lines. [12/10] (https://github.com/bbatsov/ruby-style-guide#short-methods)